```bash
OutDir=chromosome_alignment_circos/589_vs_580_circos
  mkdir -p $OutDir
```
```bash
 QueryGenome=$(ls chromosome_files/*/589/589_all.fa)
  ProgDir=~/git_repos/tools/seq_tools/circos
  python $ProgDir/fasta2circos.py --genome $QueryGenome --contig_prefix "589_" > $OutDir/query_genome.txt

  Ref_genome=$(ls assembly/misc_publications/P.stipitis/Y-11545_v2/pichia.allmasked)
  ProgDir=~/git_repos/tools/seq_tools/circos
  python $ProgDir/fasta2circos.py --genome $Ref_genome --contig_prefix "Y-11545_" > $OutDir/ref_genome.txt

  cat $OutDir/query_genome.txt > $OutDir/query_ref_genome.txt
  tac $OutDir/ref_genome.txt >> $OutDir/query_ref_genome.txt
```
```bash
OutDir=chromosome_alignment_circos/589_vs_580_circos
Coords=$(ls analysis/genome_alignment/mummer/chromosomes/P.stipitis/589/589_vs_Y-11454_v2/589_vs_Y-11454_v2_coords.tsv)
ProgDir=/projects/oldhome/armita/git_repos/emr_repos/scripts/alternaria/pathogen/genome_alignment
$ProgDir/nucmer_coords2circos.py --inp_coords $Coords --queery_id 589 --ref_id Y-11545 > $OutDir/query_vs_ref_links.txt
cat $OutDir/query_vs_ref_links.txt > $OutDir/query_vs_ref_links_edited.txt
```
A file showing contig orientations was made:

```bash
cat $OutDir/query_ref_genome.txt | cut -f3 -d ' ' | sed "s/$/; /g" | tr -d '\n' > $OutDir/query_contig_order.txt
  ProgDir=/projects/oldhome/armita/git_repos/emr_repos/tools/seq_tools/circos
  $ProgDir/find_contig_orientation.py --links_file $OutDir/query_vs_ref_links_edited.txt > $OutDir/query_vs_ref_contig_orientation.txt
```
Contig order was selected by taking the first line of that file and then also taking the reversed order of contigs using the command:

```bash
cat $OutDir/query_vs_ref_contig_orientation.txt | grep -A1 'Order of all seen contigs' | tail -n1 | sed "s/, /\n/g" > tmp.txt
cat $OutDir/query_ref_genome.txt | grep 'Y-11545' | grep -w -v -f tmp.txt | cut -f3 -d ' '| tr -d '\n' | sed 's/Y-11545/, Y-11545/g'
# cat $OutDir/Ag_As_genome.txt | grep 'Ag' | cut -f3 -d ' ' | tr -d '\n' | sed 's/Ag/, Ag/g' >> tmp.txt

echo "Order of unseen 589 contigs and remaining 589 contigs"
cat $OutDir/query_ref_genome.txt | grep -w -v -f tmp.txt | cut -f3 -d ' '| tr -d '\n' | sed 's/589/, 589/g' | sed 's/Y-11545/, Y-11545/g'
```
589_589_Chromosome_1; 589_589_Chromosome_2; 589_589_Chromosome_3; 589_589_Chromosome_4; 589_589_Chromosome_5; 589_589_Chromosome_6; 589_589_Chromosome_7; 589_589_Chromosome_8; Y-11545_chr_8.1; Y-11545_chr_7.1; Y-11545_chr_6.1; Y-11545_chr_5.1; Y-11545_chr_4.1; Y-11545_chr_3.1; Y-11545_chr_2.1; Y-11545_chr_1.2; Y-11545_chr_1.1;

