OutDir=chromosome_alignment_circos/589_vs_591_circos
  mkdir -p $OutDir
```
```bash
 QueryGenome=$(ls chromosome_files/*/591/591_all.fa)
  ProgDir=~/git_repos/tools/seq_tools/circos
  python $ProgDir/fasta2circos.py --genome $QueryGenome --contig_prefix "591_" > $OutDir/query_genome.txt

  Ref_genome=$(ls chromosome_files/*/589/589_all.fa)
  ProgDir=~/git_repos/tools/seq_tools/circos
  python $ProgDir/fasta2circos.py --genome $Ref_genome --contig_prefix "589_" > $OutDir/ref_genome.txt

  cat $OutDir/query_genome.txt > $OutDir/query_ref_genome.txt
  tac $OutDir/ref_genome.txt >> $OutDir/query_ref_genome.txt
```
```bash
OutDir=chromosome_alignment_circos/589_vs_591_circos
Coords=$(ls analysis/genome_alignment/mummer/chromosomes/P.stipitis/591/591_vs_589/591_vs_589_coords.tsv)
ProgDir=/projects/oldhome/armita/git_repos/emr_repos/scripts/alternaria/pathogen/genome_alignment
$ProgDir/nucmer_coords2circos.py --inp_coords $Coords --queery_id 591 --ref_id 589 > $OutDir/query_vs_ref_links.txt
cat $OutDir/query_vs_ref_links.txt > $OutDir/query_vs_ref_links_edited.txt
```
A file showing contig orientations was made:

```bash
cat $OutDir/query_ref_genome.txt | cut -f3 -d ' ' | sed "s/$/; /g" | tr -d '\n' > $OutDir/query_contig_order.txt
  ProgDir=/projects/oldhome/armita/git_repos/emr_repos/tools/seq_tools/circos
  $ProgDir/find_contig_orientation.py --links_file $OutDir/query_vs_ref_links_edited.txt > $OutDir/query_vs_ref_contig_orientation.txt
```
Contig order was selected by taking the first line of that file and then also taking the reversed order of contigs using the command:

```bash
cat $OutDir/query_vs_ref_contig_orientation.txt | grep -A1 'Order of all seen contigs' | tail -n1 | sed "s/, /\n/g" > tmp.txt
cat $OutDir/query_ref_genome.txt | grep '589' | grep -w -v -f tmp.txt | cut -f3 -d ' '| tr -d '\n' | sed 's/589/, 589/g'
# cat $OutDir/Ag_As_genome.txt | grep 'Ag' | cut -f3 -d ' ' | tr -d '\n' | sed 's/Ag/, Ag/g' >> tmp.txt

echo "Order of unseen 591 contigs and remaining 591 contigs"
cat $OutDir/query_ref_genome.txt | grep -w -v -f tmp.txt | cut -f3 -d ' '| tr -d '\n' | sed 's/591/, 591/g' | sed 's/589/, 589/g'
```
```
, 591_, 591_Chromosome_1, 591_, 591_Chromosome_2, 591_, 591_Chromosome_3, 591_, 591_Chromosome_4, 591_, 591_Chromosome_5, 591_, 591_Chromosome_6, 591_, 591_Chromosome_7, 591_, 591_Chromosome_8, 589_, 589_Chromosome_8, 589_, 589_Chromosome_7, 589_, 589_Chromosome_6, 589_, 589_Chromosome_5, 589_, 589_Chromosome_4, 589_, 589_Chromosome_3, 589_, 589_Chromosome_2, 589_, 589_Chromosome_1
```
```
OutDir=/projects/oldhome/groups/harrisonlab/project_files/Pichia/chromosome_alignment_circos/589_vs_591_circos
ProgDir=/projects/oldhome/vegasa/git_repos/scripts/pichia/genome_alignment/Chromosome_level_assembly/589_vs_591
circos -conf $ProgDir/589_vs_591_circos.conf -outputdir $OutDir
mv $OutDir/circos.png $OutDir/589_vs_591_circos.png
mv $OutDir/circos.svg $OutDir/589_vs_591_circos.svg
```




